src/Nethermind/Nethermind.Evm/TransactionProcessing/TransactionProcessor.cs

```diff
@@ -444,7 +449,7 @@ protected virtual TransactionResult IncrementNonce(Transaction tx, BlockHeader h
             out long spentGas,
             out byte statusCode)
         {
-            bool validate = !opts.HasFlag(ExecutionOptions.NoValidation);
+            bool validate = !opts.HasFlag(ExecutionOptions.NoValidation) || opts.HasFlag(ExecutionOptions.Simulation);
```

Comment:
instead of adding `ExecutionOptions.Simulation`, can we use some override in SimulateTransactionProcessor? I would like to avoid bloating that enum.

Reply:
Refactored, please take a look.